hstrat/hstrat/__init__.py

```diff
@@ -1,13 +1,62 @@
 """Convenient library repack as a flat namespace."""
 
 from .._auxiliary_lib import get_hstrat_version
-from ..frozen_instrumentation import *  # noqa: F401
-from ..genome_instrumentation import *  # noqa: F401
-from ..juxtaposition import *  # noqa: F401
-from ..phylogenetic_inference import *  # noqa: F401
-from ..serialization import *  # noqa: F401
-from ..stratum_retention_strategy import *  # noqa: F401
-from ..stratum_retention_viz import *  # noqa: F401
-from ..test_drive import *  # noqa: F401
+
+from .. import (
+    frozen_instrumentation as __frozen_instrumentation,
+    genome_instrumentation as __genome_instrumentation,
+    juxtaposition as __juxtaposition,
+    phylogenetic_inference as __phylogenetic_inference,
+    serialization as __serialization,
+    stratum_retention_strategy as __stratum_retention_strategy,
+    stratum_retention_viz as __stratum_retention_viz,
+    test_drive as __test_drive,
+)
+
+
+from ..frozen_instrumentation import __all__ as __frozen_instrumentation_all__
+from ..genome_instrumentation import __all__ as __genome_instrumentation_all__
+from ..juxtaposition import __all__ as __juxtaposition_all__
+from ..phylogenetic_inference import __all__ as __phylogenetic_inference_all__
+from ..serialization import __all__ as __serialization_all__
+from ..stratum_retention_strategy import (
+    __all__ as __stratum_retention_strategy_all__,
+)
+from ..stratum_retention_viz import __all__ as __stratum_retention_viz_all__
+from ..test_drive import __all__ as __test_drive_all__
+
+
+_DIR = [  # determined with from hstrat import hstrat; dir(hstrat)
+    "__all__",
+    "__builtins__",
+    "__cached__",
+    "__doc__",
+    "__file__",
+    "__getattr__",
+    "__loader__",
+    "__name__",
+    "__package__",
+    "__path__",
+    "__spec__",
+    "__version__",
+]
 
 __version__ = get_hstrat_version()
+
+
+def __getattr__(name: str) -> None:
+    if name in _DIR:
```

Comment:
can we just flip the order here? Try to get the name out of one of the __all__'s and, if that fails fall back to `eval(name)`.

Reply:
Wouldn't that potentially cause attributes that one could want from the flat namespace (such as __doc__, or __all__) to be retrieved from another module?

Reply:
I think (but not sure) __getattr__ is last resort. So if those might be handled already and not go into this func

Reply:
ha markdown is turning the dunders into bold